lib/flexible_enum/name_configurator.rb

```diff
@@ -3,6 +3,11 @@ class NameConfigurator < AbstractConfigurator
     def apply
       configurator = self
 
+      add_instance_method("#{attribute_name}_options") do
```

Comment:
`name_configurator.rb` contains logic for adding methods that expose the name variants of the current enum value. I think this new functionality isn't really about names. What do you think about either moving the options addition here to a new file or renaming `name_configurator.rb` to `identity_configurator.rb` (or similar) to better indicate that this file is about more than just names?